### DISCUSSION TITLE
[RFC] Auth UI — Template based examples - deprecate @supabase/auth-ui-[*] packages

### DISCUSSION THREAD

**Discussion by MildTomato** [supabase/supabase#19863] [General]
This is in regards to the [Auth UI packages](https://github.com/supabase/auth-ui).

I really like the shift to "copy and paste" templates for components. [@shadcn](https://twitter.com/shadcn) and [@authkit](https://twitter.com/authkit) are setting a new standard for developing an app fast without installing loads of deps.

I think this exposes the main problem with Supabase Auth UI (and other UI based deps); that they are a _dependency_; rather than a quick launchpad to get something working. Everything becomes trapped inside the dependency; such as the styling/CSS and customization become hard to change and adapt into whatever your design system becomes.

So, should [@supabase](https://twitter.com/supabase) Auth UI adopt a template approach? Everything is then exposed and editable by the developer to mould into their app how they see fit.

The main proposal would be:

- Deprecate the auth-ui packages (@supabase/auth-ui-react, @supabase/auth-ui-solid, @supabase/auth-ui-svelte)
- We use radix (perhaps take a lot of inspiration from shadcn/ui - use same component structures)
- We use tailwind as standard; Developers can choose to use something else, it should be easy to delete the tailwind from the templates and use something else.
- TailwindCSS is starting to become a default in web apps so adopt it as a default.
- We make copy and paste page examples for the most common use cases
- We priotize making it **very** easy to copy and paste a few files to make everything '_just work_' with Supabase.
- We make use of framework ports of radix/shadcn like https://github.com/radix-vue/shadcn-vue + https://www.radix-vue.com/

It would end up looking a bit like self hosted AuthKit. And perhaps maybe in time we can provide interactive docs that runs the examples, with code ready to copy like [tailwindui](https://tailwindui.com).

Screenshots from AuthKit repo:
<img width="520" alt="Screenshot 2023-12-19 at 5 17 46 PM" src="https://github.com/supabase/supabase/assets/8291514/2618315a-c63d-4373-9404-666ed495919a">

---

This rough RFC was based on a tweet of mine a week ago
original tweet from @MildTomato :
<img width="520" alt="Screenshot 2023-12-19 at 5 16 02 PM" src="https://github.com/supabase/supabase/assets/8291514/19930880-3f59-499d-a798-0895802411e8">
https://x.com/JSummersMuir/status/1732800183514325374?s=20

**Comment by zernonia** [comment]
I think this is a brilliant move! Exposing as editable component allow dev to choose which Auth component template they want and further customize it. 

I believe this also lower down the barrier to support other framework, as most of the popular lib has their own primitive library now.

Let me know if need help with porting to Vue. Glad to contribute to @supabase again!!!

**Comment by zplzpl** [comment]
Any new developments?
